README.md

```diff
@@ -1,2 +1,26 @@
-# Intelligent_placer
-University assignment: python lib that check if items in photo can be placed into figure nearby items
+# Intelligent placer
+Given a picture of several objects on a light horizontal surface and a polygon as input, determine whether it is possible to place all these objects simultaneously on the plane so that they fit into the polygon.
+
+## Task definition
+Algorithm input: 
+- The path to a file with tif, png or jpg extension containing a three-channel RGB image on which on the horizontal light surface there is a sheet of A4 paper with a polygon drawn on the left and objects from a known set on the right;
+- The path to the file with the same extension as in the previous paragraph, which contains an image of the surface on which a sheet of paper with a polygon and items will be placed;
+- 4 integers describing the area within which the integer objects we are going to consider are contained. the first and second numbers represent the start and end pixel (inclusive) horizontally, and the second and third numbers represent the start and end pixel of the area vertically.
+Algorithm output:
+- True if goods can be placed in a polygon, otherwise false.
+
+## Input image requirements
+- Objects cannot intersect;
+- There must be a distance of at least 10 pixels between objects;
+- The border of a polygon must be at least 4 pixels thick at its narrowest point;
```

Comment:
I didn't write about it myself, but it might be worth considering. If there are no borders on the upper value of the width of the polygon border, then, with a particularly wide border, it can be recognized as not a border but an object? Then, for example, your algorithm will not find a polygon? Or is it assumed that the first object on the left is being searched for, and it is automatically recognized as a polygon?

Reply:
I'm not sure I understood the question. A polygon will not be found if its boundary has less than 4 pixels. The polygon will most likely be searched by the position of the sheet of paper: it can only be drawn on it. If it has a wide border, or even completely filled in, it will not be affected, since the outer boundary is considered. But thanks for the comment, it gave me the idea to add a few conditions to the problem definition